langchain4j/src/main/java/dev/langchain4j/memory/chat/SummarizedTokenWindowChatMemory.java

```diff
@@ -0,0 +1,260 @@
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Generates a new summary given an existing summary and a new message. Only supports UserMessage and AiMessage without
+     * tool execution requests/responses for now.
+     * @param messages the list of messages in the chat memory.
+     * @param evictedMessages the list of ChatMessages to be evicted.
+     */
+    private void addEvictedMessagesToSummary(List<ChatMessage> messages, List<ChatMessage> evictedMessages) {
+        StringBuilder messagesToSummarize = new StringBuilder();
+
+        for (ChatMessage evictedMessage : evictedMessages) {
+            switch (evictedMessage.type()) {
+                case USER:
+                    messagesToSummarize.append("User: ").append(((UserMessage) evictedMessage).singleText()).append("\n");
+                    break;
+                case AI:
+                    if (((AiMessage) evictedMessage).hasToolExecutionRequests()) {
+                        log.warn("Tool execution requests are not supported in this method");
+                    } else {
+                        messagesToSummarize.append("AI: ").append(((AiMessage) evictedMessage).text()).append("\n");
+                    }
+                    break;
+            }
+        }
+
+        Optional<SystemMessage> maybeSystemMessage = findSystemMessage(messages);
+        if (maybeSystemMessage.isPresent()) {
+            // Not supporting overwriting system messages for now
+            String existingSummary = extractSummary(maybeSystemMessage.get()).orElse("");
+            String newSummary = generateMessageSummary(messagesToSummarize.toString(), existingSummary);
+            SystemMessage updatedSummarizedSystemMessage = generateSummarizedSystemMessage(maybeSystemMessage.get().text(), newSummary);
+
+            // Replace the existing system message with the updated one
+            int systemMessageIndex = messages.indexOf(maybeSystemMessage.get());
+            if (systemMessageIndex != -1) {
+                messages.set(systemMessageIndex, updatedSummarizedSystemMessage);
+            } else {
+                // If the system message is not found in the list (which shouldn't happen),
+                // add the updated system message to the beginning of the list
+                messages.add(0, updatedSummarizedSystemMessage);
+            }
+        }
+    }
+
+    /**
+     * Converts the original system message into a SummarizedSystemPrompt with placeholder values for the original
+     * message and the summary.
+     * @param originalSystemMessage
+     * @return
+     */
+    private SystemMessage generateSummarizedSystemMessage(SystemMessage originalSystemMessage) {
+        SummarizedSystemPrompt summarizerPrompt = new SummarizedSystemPrompt(originalSystemMessage.text(), "none, the conversation has just begun.");
+        Prompt prompt = StructuredPromptProcessor.toPrompt(summarizerPrompt);
+        return prompt.toSystemMessage();
```

Comment:
It could be replaced with:

```java
return generateSummarizedSystemMessage(originalSystemMessage.text(), "none, the conversation has just begun.");
```